### PR TITLE
Add org.nongnu.enigma

### PR DESCRIPTION
**Pull request by MrTarantoga**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://lists.nongnu.org/archive/html/enigma-devel/2021-09/msg00000.html
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by MrTarantoga** [comment]
bot, build org.nongnu.enigma

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply flathub.json:2]: Well, currently it does not compile on arm (only tested with github actions). Also I have no device where I can test it in a productiv case.
On the other side it was originally only developed for x86 platform without the goal of portability to other cpu architectures.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply flathub.json:2]: arm or aarch64?

also github action is know to fail with NOT x86_64, flathub is different.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.nongnu.enigma.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply org.nongnu.enigma.metainfo.xml:1]: I'm currently preparing the patch and the pull request.

File: org.nongnu.enigma.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      - type: git
+        url: https://github.com/ImageMagick/ImageMagick.git
+        tag: "7.1.0-5"
+        commit: 74f8154136d2f631ca3e0421e6cf6c8dccd548f7
+
+  - name: Enigma-Game
+    no-autogen: false
+    sources:
+      - type: git
+        url: https://github.com/MrTarantoga/Enigma.git
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply org.nongnu.enigma.yml:64]: The pull request was not merged to the main repository. Its mentioned thankful in the [mailing list ](https://lists.nongnu.org/archive/html/enigma-devel/2021-09/msg00000.html) but I can not explain why it is not merged.

**Comment by MrTarantoga** [comment]
aarch64 was the tested platform.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Enigma
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply README.md:1]: It is description of the branch. In [App Maintenance](https://github.com/flathub/flathub/wiki/App-Maintenance#optional) README.md is mentioned as "whatever information you see fit". My intention was to give a short description of the branch to know what it contain.

File: flathub.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply flathub.json:2]: I can remove the tag and build it once on aarch64.

File: org.nongnu.enigma.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+    post-install:
+      - install -d /app/extra/export/share/applications
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - mv /app/share/applications/enigma.desktop /app/share/applications/org.nongnu.enigma.desktop
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/org.nongnu.enigma.png
+      - sed -i 's/Icon=enigma/Icon=org.nongnu.enigma.png/g' /app/share/applications/org.nongnu.enigma.desktop
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply org.nongnu.enigma.yml:71]: No it has no svg icon in upstream.

File: org.nongnu.enigma.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  - --socket=pulseaudio
+  # OpenGL redering
+  - --device=dri
+
+modules:
+  - name: texi2html
+    sources:
+      - type: archive
+        url: https://download.savannah.gnu.org/releases/texi2html/texi2html-5.0.tar.gz
+        sha256: e60edd2a9b8399ca615c6e81e06fa61946ba2f2406c76cd63eb829c91d3a3d7d
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:23]:
```suggestion
        sha256: e60edd2a9b8399ca615c6e81e06fa61946ba2f2406c76cd63eb829c91d3a3d7d
      - type: shell
        commands:
          - cp -p /usr/share/automake-*/config.{sub,guess} .
```

You need to add this for building on other arch.

File: flathub.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply flathub.json:2]: see my other comment for the patch change you need apply.

File: org.nongnu.enigma.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply org.nongnu.enigma.yml]: Changed in [commit](https://github.com/flathub/flathub/pull/2653/commits/e61c0c5517d9c1091600f7c287e57e3b3bcd32d9).

File: org.nongnu.enigma.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        url: https://github.com/MrTarantoga/Enigma.git
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+    post-install:
+      - install -d /app/extra/export/share/applications
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - mv /app/share/applications/enigma.desktop /app/share/applications/org.nongnu.enigma.desktop
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/org.nongnu.enigma.png
+      - sed -i 's/Icon=enigma/Icon=org.nongnu.enigma.png/g' /app/share/applications/org.nongnu.enigma.desktop
+
+  - name: AppStream.xml
```
[the remark this answers is not in the document]
Comment (MrTarantoga) [reply org.nongnu.enigma.yml]: Changed in [commit](https://github.com/flathub/flathub/pull/2653/commits/6cd6842f0cc6d04d12740f4cfcfc5b87a1c84938).

File: org.nongnu.enigma.yml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+      - type: git
+        url: https://github.com/MrTarantoga/Enigma.git
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+      - type: file
+        path: org.nongnu.enigma.metainfo.xml
+    post-install:
+      - install -d /app/extra/export/share/applications
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/enigma.png
+      - sed -i 's/Icon=enigma/Icon=${FLATPAK_DEST}/g' /app/share/applications/enigma.desktop
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:75]:
```suggestion
```
This is taken care of byt the `rename-icon`

File: org.nongnu.enigma.yml [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        commit: 74f8154136d2f631ca3e0421e6cf6c8dccd548f7
+
+  - name: Enigma-Game
+    no-autogen: false
+    sources:
+      - type: git
+        url: https://github.com/MrTarantoga/Enigma.git
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+      - type: file
+        path: org.nongnu.enigma.metainfo.xml
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:70]: you still need to install it

File: org.nongnu.enigma.yml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    post-install:
+      - install -d /app/extra/export/share/applications
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:72]: why do you need this?

File: org.nongnu.enigma.yml [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/enigma.png
+      - sed -i 's/Icon=enigma/Icon=${FLATPAK_DEST}/g' /app/share/applications/enigma.desktop
+
+rename-icon: enigma
+rename-desktop-file: enigma.desktop
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:78]: please put these toward the top like at line 7 (where they are usually)

it's not required but I almost missed them.

File: org.nongnu.enigma.yml [hunk 12]
```diff
@@ -0,0 +1,83 @@
+app-id: org.nongnu.enigma
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+appstream-compose: false
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:5]:
```suggestion
```

Leave this as default (`true`). It will do the right thing

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by MrTarantoga** [comment]
bot, build org.nongnu.enigma

File: org.nongnu.enigma.yml (hunk 9 above)
Comment (hfiguiere) [reply org.nongnu.enigma.yml:70]: this is why the build failed...

File: org.nongnu.enigma.yml (hunk 11 above)
Comment (MrTarantoga) [reply org.nongnu.enigma.yml]: See [commit](https://github.com/flathub/flathub/pull/2653/commits/eaa1eff62c74fe1ed6600e23c6e4bae9b3f161d2)

File: org.nongnu.enigma.yml (hunk 10 above)
Comment (MrTarantoga) [reply org.nongnu.enigma.yml]: [Removed ](https://github.com/flathub/flathub/pull/2653/commits/9856f0b47f6c5b4ed4a4e44b43c26db464cb40b2), remnant of a past experiment

File: org.nongnu.enigma.yml (hunk 9 above)
Comment (MrTarantoga) [reply org.nongnu.enigma.yml:71]: Done, in [commit](https://github.com/flathub/flathub/pull/2653/commits/f52418aeceeecd99375a557e5b0d5b86bb9c3dc8)

File: org.nongnu.enigma.yml [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+      - type: git
+        url: https://github.com/MrTarantoga/Enigma.git
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+      - type: file
+        path: org.nongnu.enigma.metainfo.xml
+    post-install:
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/enigma.png
+      - appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:76]:
```suggestion
```

You do not need this.

**Comment by MrTarantoga** [comment]
bot, build org.nongnu.enigma

File: org.nongnu.enigma.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (barthalion) [reply org.nongnu.enigma.yml:68]: Can you please use upstream source and apply your patches as part of the manifest instead?

File: org.nongnu.enigma.yml [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        url: https://github.com/MrTarantoga/Enigma.git
+        commit: 0db9df0aea0c7a874a494647210265340425bdd9
+      - type: file
+        path: org.nongnu.enigma.metainfo.xml
+    post-install:
+      - install -d /app/share/icons/hicolor/128x128/apps/
+      - install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
+      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/enigma.png
+
+cleanup:
```
Comment (barthalion) [inline org.nongnu.enigma.yml:77]: By convention, a global `cleanup` section is usually above `modules`.

Comment (MrTarantoga) [reply org.nongnu.enigma.yml]: Fixed in [commit](https://github.com/flathub/flathub/pull/2653/commits/dccee3916d5c05efa9ecbe2e15b401dc15c10585)

**Comment by MrTarantoga** [comment]
bot, build org.nongnu.enigma

**Pull request closed without merging**